Crash when supplying input

<!--
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

If your issue relates to a specific Terraform provider, please open it in the provider's own repository. The index of providers is at https://github.com/terraform-providers .
-->

### Terraform Version
<!---
Run `terraform -v` to show the version, and paste the result between the ``` marks below.

If you are not running the latest version of Terraform, please try upgrading because your issue may have already been fixed.
-->

```
0.12.20
```

### Crash Output
<!--
If the console output indicates that Terraform crashed, please share a link to a GitHub Gist containing the output of the `crash.log` file.
-->

```
Initializing modules...

Initializing the backend...
Backend configuration changed!

Terraform has detected that the configuration specified for the backend
has changed. Terraform will now check for existing state in the backends.


organization
  The name of the organization containing the targeted workspace(s).

  Enter a value: ^C
panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 1 [running]:
github.com/zclconf/go-cty/cty.Value.GetAttr(0x2e5fde0, 0xc00057d9b0, 0x0, 0x0, 0x29cd089, 0xc, 0x1077c12, 0x0, 0x2e1b740, 0x0)
	/private/tmp/terraform-20200122-15112-10ii3r0/terraform-0.12.20/pkg/mod/github.com/zclconf/go-cty@v1.2.1/cty/value_ops.go:711 +0x408
github.com/hashicorp/terraform/backend/remote.(*Remote).PrepareConfig(0xc000220fd0, 0x2e5fde0, 0xc00057d9b0, 0x0, 0x0, 0x1, 0xc000ec28b0, 0x1, 0x1, 0x3d96c00, ...)
	/private/tmp/terraform-20200122-15112-10ii3r0/terraform-0.12.20/src/github.com/hashicorp/terraform/backend/remote/backend.go:146 +0x7f
github.com/hashicorp/terraform/command.(*Meta).backendInitFromConfig(0xc0004b1c00, 0xc001430960, 0xc7, 0xc0002a6270, 0xc7, 0xc0002a6000, 0xce, 0x0, 0x0, 0xc001707270, ...)
	/private/tmp/terraform-20200122-15112-10ii3r0/terraform-0.12.20/src/github.com/hashicorp/terraform/command/meta_backend.go:1116 +0x2de
github.com/hashicorp/terraform/command.(*Meta).backend_C_r_S_changed(0xc0004b1c00, 0xc001430960, 0xf5502056, 0xc00097cae0, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/private/tmp/terraform-20200122-15112-10ii3r0/terraform-0.12.20/src/github.com/hashicorp/terraform/command/meta_backend.go:876 +0xa2
github.com/hashicorp/terraform/command.(*Meta).backendFromConfig(0xc0004b1c00, 0xc001707700, 0x0, 0x0, 0x0, 0x0, 0x0)
	/private/tmp/terraform-20200122-15112-10ii3r0/terraform-0.12.20/src/github.com/hashicorp/terraform/command/meta_backend.go:551 +0x8eb
github.com/hashicorp/terraform/command.(*Meta).Backend(0xc0004b1c00, 0xc001707700, 0x0, 0xc001707600, 0x1, 0x1, 0x0)
	/private/tmp/terraform-20200122-15112-10ii3r0/terraform-0.12.20/src/github.com/hashicorp/terraform/command/meta_backend.go:85 +0x599
github.com/hashicorp/terraform/command.(*InitCommand).initBackend(0xc0004b1c00, 0xc00048a100, 0x29d07e5, 0xf, 0xc000546f40, 0x1, 0x0, 0x0, 0x0, 0x0, ...)
	/private/tmp/terraform-20200122-15112-10ii3r0/terraform-0.12.20/src/github.com/hashicorp/terraform/command/init.go:482 +0x1bc
github.com/hashicorp/terraform/command.(*InitCommand).Run(0xc0004b1c00, 0xc0000ba030, 0x0, 0x0, 0xc000575830)
	/private/tmp/terraform-20200122-15112-10ii3r0/terraform-0.12.20/src/github.com/hashicorp/terraform/command/init.go:287 +0x1889
github.com/mitchellh/cli.(*CLI).Run(0xc000359b80, 0xc000359b80, 0xc00052fcc0, 0x1)
	/private/tmp/terraform-20200122-15112-10ii3r0/terraform-0.12.20/pkg/mod/github.com/mitchellh/cli@v1.0.0/cli.go:255 +0x1da
main.wrappedMain(0x0)
	/private/tmp/terraform-20200122-15112-10ii3r0/terraform-0.12.20/src/github.com/hashicorp/terraform/main.go:239 +0xc43
main.realMain(0x0)
	/private/tmp/terraform-20200122-15112-10ii3r0/terraform-0.12.20/src/github.com/hashicorp/terraform/main.go:103 +0xb4
main.main()
	/private/tmp/terraform-20200122-15112-10ii3r0/terraform-0.12.20/src/github.com/hashicorp/terraform/main.go:39 +0x3a
```


Configs
```hcl
terraform {
  backend "remote" {
  }
}
```

### Steps to Reproduce
<!--
Please list the full steps required to reproduce the issue, for example:
1. `terraform init`
2. `terraform apply`
-->
Run terraform init, when asked for organization name hit Ctrl+C

--- a/backend/remote/backend.go
+++ b/backend/remote/backend.go
@@ -142,6 +142,9 @@ func (b *Remote) ConfigSchema() *configschema.Block {
 // PrepareConfig implements backend.Backend.
 func (b *Remote) PrepareConfig(obj cty.Value) (cty.Value, tfdiags.Diagnostics) {
 	var diags tfdiags.Diagnostics
+	if obj.IsNull() {
+		return obj, diags
+	}
 
 	if val := obj.GetAttr("organization"); val.IsNull() || val.AsString() == "" {
 		diags = diags.Append(tfdiags.AttributeValue(
@@ -188,6 +191,9 @@ func (b *Remote) PrepareConfig(obj cty.Value) (cty.Value, tfdiags.Diagnostics) {
 // Configure implements backend.Enhanced.
 func (b *Remote) Configure(obj cty.Value) tfdiags.Diagnostics {
 	var diags tfdiags.Diagnostics
+	if obj.IsNull() {
+		return diags
+	}
 
 	// Get the hostname.
 	if val := obj.GetAttr("hostname"); !val.IsNull() && val.AsString() != "" {

--- a/backend/remote/backend_test.go
+++ b/backend/remote/backend_test.go
@@ -123,6 +123,9 @@ func TestRemote_config(t *testing.T) {
 			}),
 			valErr: `Only one of workspace "name" or "prefix" is allowed`,
 		},
+		"null config": {
+			config: cty.NullVal(cty.EmptyObject),
+		},
 	}
 
 	for name, tc := range cases {

